packages/fxa-shared/l10n/parseAcceptLanguage.ts

```diff
@@ -0,0 +1,71 @@
+import { negotiateLanguages } from '@fluent/langneg';
+import availableLocales from './supportedLanguages.json';
+import { EN_GB_LOCALES } from './otherLanguages';
+
+/**
+ * Takes an acceptLanguage value (assumed to come from an http header) and returns
+ * a set of valid locales in order of precendence.
+ * @param acceptLanguage - Http header accept-language value
+ * @param supportedLanguages - List of supported language codes
+ * @returns A list of associated supported locales. If there are no matches for the given
+ *          accept langauge, then the default locle, en, will be returned.
+ *
+ */
+export function parseAcceptLanguage(
+  acceptLanguage: string,
+  supportedLanguages?: string[]
+) {
+  if (!supportedLanguages) {
+    supportedLanguages = availableLocales;
+  }
+  if (acceptLanguage == null) {
+    acceptLanguage = '';
+  }
```

Comment:
I think you can give these defaults in the params instead unless there's a reason to explicitly check here, like:
```
export function parseAcceptLanguage(
  acceptLanguage = '',
  supportedLanguages?: string[] = availableLocales
) {
```

Reply:
You would think... but since we are calling this from java script in auth-server, and there's no type safety there, I was hitting edge cases where undefined was getting sent in. 

As for 'en' vs ''. The language negotiation already declares en as the default, so in my opinion it's less repetitive to set this to an empty string, and then just declare the default in a single spot when language negotiation happens.

Reply:
> You would think... but since we are calling this from javascript in auth-server, and there's no type safety there, I was hitting edge cases where undefined was getting sent in.

Huh, if `undefined` is passed in for `acceptLanguage`, wouldn't it get set to `''` and if `supportedLanguages` is `undefined` wouldn't it get set to `availableLocales`? I might just not be understanding, either way what you have seems fine. 🙂

Yeah, disregard that, I edited my comment and agree.